### DISCUSSION TITLE
mini.ai plugin seems to be never loaded.

### DISCUSSION THREAD

**Discussion by dompuiu** [LazyVim/LazyVim#155] [Q&A] [answered]
It seems that the `mini.ai` plugin is not loading with the current setup. 

I have a fresh Neovim config with the code from the [starter](https://github.com/LazyVim/starter) project.

When I open NeoVim, the `mini.ai` plugin appears at not being loaded which is the expected behavior. I check the status by going to the  lazy.nvim main screen. I open a file and start editing it. I do some text-objects operations. When I got to lazy.nvim, the plugin is still appearing in the not loaded plugin list.

To make it load I created a new plugin file with the content:

```
return {
  "echasnovski/mini.ai",
  lazy = false,
}
```

Am I am doing something wrong or is this a bug?

Thanks for all the work you are doing!

**Comment by luxus** [comment]
```lua
 -- better text-objects
  {
    "echasnovski/mini.ai",
    keys = {
      { "a", mode = { "x", "o" } },
      { "i", mode = { "x", "o" } },
    },
    dependencies = {
      {
        "nvim-treesitter/nvim-treesitter-textobjects",
        init = function()
          -- no need to load the plugin, since we only need its queries
          require("lazy.core.loader").disable_rtp_plugin "nvim-treesitter-textobjects"
        end,
      },
    },
    opts = function()
      local ai = require "mini.ai"
      return {
        n_lines = 500,
        custom_textobjects = {
          o = ai.gen_spec.treesitter({
            a = { "@block.outer", "@conditional.outer", "@loop.outer" },
            i = { "@block.inner", "@conditional.inner", "@loop.inner" },
          }, {}),
          f = ai.gen_spec.treesitter({ a = "@function.outer", i = "@function.inner" }, {}),
          c = ai.gen_spec.treesitter({ a = "@class.outer", i = "@class.inner" }, {}),
        },
      }
    end,
    config = function(_, opts)
      local ai = require "mini.ai"
      ai.setup(opts)
    end,
  },
```
try this

**Reply by dompuiu** [reply]
The code you pasted seems to be identical with this one: https://github.com/LazyVim/LazyVim/blob/aa47cc862b95b7e4fd7f1d1359b4f123e3080dce/lua/lazyvim/plugins/coding.lua#L144-L177. Which doesn't seem to work for me. 

By creating the new plugin file and setting `lazy = false`, I am telling `lazy.nvim` to load the `echasnovski/mini.ai` plugin from the start. It seems to me that I should not set the `lazy` flag to `false` and maybe I am missing something else.

**Reply by luxus** [reply]
sorry u are right.. the plugin doesn't load

**Reply by luxus** [reply]
you forgot the config part.

```lua
-- better text-objects
  {
    "echasnovski/mini.ai",
    lazy = false,
    keys = {
      { "a", mode = { "x", "o" } },
      { "i", mode = { "x", "o" } },
      { "g", mode = { "x", "o" } },
    },
    dependencies = {
      {
        "nvim-treesitter/nvim-treesitter-textobjects",
        init = function()
          -- no need to load the plugin, since we only need its queries
          require("lazy.core.loader").disable_rtp_plugin "nvim-treesitter-textobjects"
        end,
      },
    },
    opts = function()
      local ai = require "mini.ai"
      return {
        mappings = {
          -- Main textobject prefixes
          around = "a",
          inside = "i",
          around_next = "an",
          inside_next = "in",
          around_last = "al",
          inside_last = "il",
          -- Move cursor to corresponding edge of `a` textobject
          goto_left = "g[",
          goto_right = "g]",
        },
        n_lines = 500,
        custom_textobjects = {
          o = ai.gen_spec.treesitter({
            a = { "@block.outer", "@conditional.outer", "@loop.outer" },
            i = { "@block.inner", "@conditional.inner", "@loop.inner" },
          }, {}),
          f = ai.gen_spec.treesitter({ a = "@function.outer", i = "@function.inner" }, {}),
          c = ai.gen_spec.treesitter({ a = "@class.outer", i = "@class.inner" }, {}),
        },
      }
    end,
    config = function(_, opts) require("mini.ai").setup(opts) end,
  },
```
I didn't get it loading via keys or VeryLazy, but this is now loading. 
dunno why its not loading via keys :(

**Reply by dompuiu** [reply]
The last config you posted has ` lazy = false,`  on the 4th line. That is why is it loading for you now.

**Reply by luxus** [reply]
yes i know, but i don't understand why it is not lazyloading.
it should load when I press a or i in visual mode or with lazy = "VeryLazy" it should load after the start as well
edit: ahh lol I *event* = "VeryLazy" works :D

**Comment by folke** [comment, marked as answer]
There is a conflict with which-key for the `a` and `i` mappings. Doing something like `daf` or `yiq`, worked, but other operator pending keymaps would not trigger `mini.ai` to load.

Fixed it by using `event = "VeryLazy"` instead.

**Reply by luxus** [reply]
thanks, yes event = "VeryLazy" works
